docs/writing-functions.md

```diff
@@ -197,6 +197,70 @@ FunctionsFramework.http "error_reporter" do |request|
 end
 ```
 
+## Shared resources
+
+Generally, functions should be self-contained and stateless, and should not use
+or share any global state in the Ruby VM. This is because serverless runtimes
+may spin up or terminate instances of your app at any time, and because a
+single instance may be running multiple functions at a time in separate threads.
+
+However, it is sometimes useful to share a resource across multiple function
+invocations that run on the same Ruby instance. For example, you might establish
+a single connection to a remote database or other service, and share it across
+function invocations, to avoid incurring the overhead of re-establishing that
```

Comment:
nit: no comma needed

share invocations to avoid X

Reply:
Fixed